### PR TITLE
php: 8.1 Support

### PR DESCRIPTION
**Pull request by JediKev**
This addresses Warnings and Fatal Errors with PHP 8.1.x. This includes:
- Fix Deprecation Warnings (Serializable)
- Fix Deprecation Warnings (line_endings)
- Fix Upgrader Issues

### CODE REVIEW

**Comment by JediKev** [comment]
**TODO:**
- [x] Fix remaining warnings with `(un)serialize`.
- [ ] Maybe address `strftime()` warnings?

File: include/class.config.php [hunk 1]
```diff
@@ -159,7 +159,7 @@ static function items($namespace, $column='namespace') {
         try {
             count($items);
         }
-        catch (InconsistentModelException $ex) {
+        catch (InconsistentModelException|mysqli_sql_exception $ex) {
```
Comment (protich) [inline include/class.config.php]: We shouldn't be catching "naked" mysql exceptions here - it needs to go to ORM and rethrown as `InconsistentModelException` expected here.

File: include/class.ostsession.php [hunk 2]
```diff
@@ -40,7 +40,7 @@ function __construct($ttl=0){
         ini_set('session.gc_maxlifetime', $ttl);
 
         // Skip db version check if version is later than 1.7
-        if (!defined('MAJOR_VERSION') && OsticketConfig::getDBVersion())
+        if (OsticketConfig::getDBVersion())
```
Comment (protich) [inline include/class.ostsession.php]: Why this change? Removing this will results in every page load checking DB version - which results in an SQL error and unnecessary query for installation after `v1.7`.

File: include/class.page.php [hunk 3]
```diff
@@ -234,10 +234,7 @@ static function lookupByType($type, $lang=false) {
         try {
             return self::objects()->filter(array('type'=>$type))->one();
         }
-        catch (DoesNotExist $ex) {
-            return null;
-        }
-        catch (InconsistentModelException $ex) {
+        catch (DoesNotExist|InconsistentModelException|mysqli_sql_exception $ex) {
```
Comment (protich) [inline include/class.page.php]: See my other comment on mysql exceptions.

**Review by protich — CHANGES_REQUESTED** [review]

**Comment by JediKev** [comment]
@protich Updated to reflect the spacing in the caught exception types (`include/class.page.php`) 👍🏻

**Review by protich — APPROVED** [review]
